aten/src/ATen/native/mps/operations/Shape.mm

```diff
@@ -638,29 +654,43 @@ void check_shape_except_dim(const Tensor &first, const Tensor &second,
           newCachedGraph = new CachedGraph(mpsGraph);
 
           // Create placeholders
-          MPSGraphTensor* inputMPSGraphTensors[inputs.size()];
-          MPSGraphTensor* castInputMPSGraphTensors[inputs.size()];
-
-          for(int i = 0; i < inputs.size(); i++) {
-            inputMPSGraphTensors[i] = mpsGraphUnrankedPlaceHolder(mpsGraph, getMPSDataType(result_type(inputs)));
-            if(getMPSDataType(result_type(inputs)) == MPSDataTypeBool)
-              castInputMPSGraphTensors[i] = [mpsGraph castTensor:inputMPSGraphTensors[i]
-                                                      toType:MPSDataTypeInt32
-                                                        name:[NSString stringWithFormat:@"inputTensor_%@", [NSNumber numberWithInt:i]]];
-            else
-              castInputMPSGraphTensors[i] = inputMPSGraphTensors[i];
+          auto len_tensor_array = inputs.size() - skipped_tensor_indices.size();
+          MPSGraphTensor* inputMPSGraphTensors[len_tensor_array];
```

Comment:
This whole logic seems more complicated then needs to be. Can we look at consolidating this. For instance you create a list of  non-skipped tensors above when creating indices and just use that instead of checkin it again.

Reply:
I simplified the logic, now we directly iterate through the non-skipped tensors here.